pkg/plugins/placementpolicy/placementpolicy.go

```
@@ -170,20 +154,28 @@ func (p *Plugin) Filter(ctx context.Context, state *framework.CycleState, pod *c
 		return framework.NewStatus(framework.Error, "failed to cast state data")
 	}
 
+	// if only "matched" in PreFilter, still a success, so continue
+	if d.status == Matched {
+		state.Write(p.getFilterStateKey(), d)
+		return framework.NewStatus(framework.Success, "")
+	}
+
 	node := nodeInfo.Node()
 	// nodeMatchesLabels is set to true if the node in the current context matches the node selector labels
 	// defined in the placement policy chosen for the pod.
-	nodeMatchesLabels := checkHasLabels(node.Labels, d.pp.Spec.NodeSelector.MatchLabels)
+	nodeMatchesLabels := checkHasLabels(node.Labels, d.nodeLabels)
 
-	// podNodePreferMatchingLabels is set to true if the pod is annotated to be on the node with matching labels
-	podNodePreferMatchingLabels := false
-	if podNodePreferMatchingLabels, err = strconv.ParseBool(pod.Annotations[v1alpha1.PlacementPolicyPreferenceAnnotationKey]); err != nil {
-		return framework.NewStatus(framework.Error, fmt.Sprintf("failed to parse pod annotation: %v", err))
+	podKey, keyError := framework.GetPodKey(pod)
+	if keyError != nil {
+		return framework.NewStatus(framework.Error, "pod key not found")
```

Let's also return the actual err as part of `framework.NewStatus`